### PR TITLE
Contest model

### PR DESCRIPTION
**Pull request by Prashant-Dhanak**
### What this PR does (required):
- It creates model, controller and routes for Contest

### Screenshots / Videos (required):
- Its a backend code so nothing is required at the moment

### Any information needed to test this feature (required):
- Contest Model created
- Contest Controllers created
- Contest Routes created and added

### Any issues with the current functionality (optional):
None yet

### CODE REVIEW

File: server/controllers/contest.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+// A list of contests that are available for submission
+exports.getContests = asyncHandler(async (req, res, next) => {
+  try {
+    const contests = await Contest.find({});
+    if (contests.length === 0) {
+      return res.status(404).json({ status: "no contests in records!!" });
+    }
+    res.status(201).json({
+      contestsFound: contests.length,
```
Comment (rajivtitus) [inline server/controllers/contest.js]: Maybe calling this number of contests would be more meaningful? :)

File: server/controllers/contest.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            deadlineDate,
+            userID
+        });
+        if (!updatedContest) {
+            return res
+                .status(404)
+                .json({ status: "contest doesn't exist in records!!" });
+        }
+        // updating contest
+        res.status(201).json({
```
Comment (rajivtitus) [inline server/controllers/contest.js]: Please use a general status code like 200.
The status code 201 is for resource creation :)

Ditto throughout!

**Review by rajivtitus — APPROVED** [review]
Great work! Just a couple of small comments 👍🏽 

In the future when you create models/controllers, you can test them using an API client (like Postman API) and maybe add some screenshots to the PR too :)

File: server/controllers/contest.js (hunk 1 above)
Comment (Prashant-Dhanak) [reply server/controllers/contest.js]: Hi,
Can you please give me some more details :), I don't think I am getting what you are trying to explain me.
Thanks

Comment (rajivtitus) [reply server/controllers/contest.js]: Just a suggestion, renaming `contestsFound` to `numberOfContests` would be more meaningful in this case :)

File: server/controllers/contest.js (hunk 2 above)
Comment (Prashant-Dhanak) [reply server/controllers/contest.js]: Sure, I did it on my latest commit.
Thanks

File: server/controllers/contest.js (hunk 1 above)
Comment (Prashant-Dhanak) [reply server/controllers/contest.js]: Oh sure, that was easy...

File: server/controllers/contest.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+const Contest = require("../models/Contest");
+const asyncHandler = require("express-async-handler");
+const ObjectId = require("mongoose").Types.ObjectId;
+
+// Given parameters passed in, create a contest
+exports.createContest = asyncHandler(async (req, res, next) => {
+    const userID = req.user.id
```
Comment (rajivtitus) [inline server/controllers/contest.js:7]: 👍🏽

**Pull request merged**